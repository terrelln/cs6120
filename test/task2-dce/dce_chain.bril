@main(a: int): int {
    b: int = const 1;
    c: int = add a b;
    d: int = id a;
    e: int = id c;
    f: int = add d e;
    g: bool = eq c f;
    h: int = add e f;
    h: int = id a;
    br g .block1 .block2;
.block1:
    g1: int = id g;
    g1: int = add g1 b;
    g1: int = add g1 g1;
    g1: int = id g1;
    g1: int = id g1;
    g1: int = id g1;
    g1: int = id g1;
    g1: int = id g;
.block2:
    g1: int = id g;
    g1: int = add g1 g1;
    g1: int = add g1 g1;
    ret a;
}