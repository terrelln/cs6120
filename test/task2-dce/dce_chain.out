@main(a: int): int {
.__block0:
  b: int = const 1;
  c: int = add a b;
  d: int = id a;
  e: int = id c;
  f: int = add d e;
  g: bool = eq c f;
  br g .block1 .block2;
.block1:
  g1: int = id g;
.block2:
  g1: int = id g;
  g1: int = add g1 g1;
  g1: int = add g1 g1;
  ret a;
}
